### PR TITLE
[python API] Change the open/save panels

### PR DESCRIPTION
**Pull request by lonnywong**
1. Open the last directory by default.
2. Remove the exception if the user cancels.

### CODE REVIEW

**Comment by gnachman** [comment]
That being said, what about returning None instead of a Result if the user cancels?

**Comment by lonnywong** [comment]
> I think the exception was useful:
> 
> * We can guarantee that the list is never empty
> * If people forget to handle it, the breakage happens immediately

1. The developer should know that the user may cancels.
2. If catch the exception, maybe a real exception happens, instead of the user cancels. It's hard to identify.

**Comment by lonnywong** [comment]
> That being said, what about returning None instead of a Result if the user cancels?

Return None is better then exception.

File: api/library/python/iterm2/iterm2/filepanel.py [hunk 1]
```diff
@@ -89,13 +89,12 @@ def message(self) -> typing.Optional[str]:
     @message.setter
     def message(self, value: str):
         self.__message = value
-        
+
     async def async_run(self, connection) -> 'iterm2.OpenPanel.Result':
```
Comment (gnachman) [inline api/library/python/iterm2/iterm2/filepanel.py:93]: The return type is now `typing.Optional['iterm2.OpenPanel.Result']`

Similar comment for save panel

File: api/library/python/iterm2/iterm2/filepanel.py [hunk 2]
```diff
@@ -106,7 +105,9 @@ async def async_run(self, connection) -> 'iterm2.OpenPanel.Result':
              f'extensions: {json.dumps(self.extensions)},' +
              f'prompt: {json.dumps(self.prompt)},' +
              f'message: {json.dumps(self.message)})'))
-        return OpenPanel.Result(response)
+        if response:
```
Comment (gnachman) [inline api/library/python/iterm2/iterm2/filepanel.py:108]: fyi I tweaked this and you'll need to pull to get it

File: api/library/python/iterm2/iterm2/filepanel.py (hunk 1 above)
Comment (lonnywong) [reply api/library/python/iterm2/iterm2/filepanel.py]: > The return type is now `typing.Optional['iterm2.OpenPanel.Result']`
> 
> Similar comment for save panel

done

**Comment by lonnywong** [comment]
@gnachman 
Is the PR ready to merge? Thanks for your time.

**Pull request merged**